openstack_controller/datadog_checks/openstack_controller/openstack_controller.py

```diff
@@ -131,39 +132,44 @@ class OpenStackControllerCheck(AgentCheck):
     def __init__(self, name, init_config, agentConfig, instances=None):
         AgentCheck.__init__(self, name, init_config, agentConfig, instances)
 
+        skip_proxy = not is_affirmative(init_config.get('use_agent_proxy', True))
+        self.proxy_config = None if skip_proxy else self.proxies
+
         self._ssl_verify = is_affirmative(init_config.get("ssl_verify", True))
-        self.keystone_server_url = init_config.get("keystone_server_url")
-        self.hypervisor_name_cache = {}
 
         self.paginated_server_limit = init_config.get('paginated_server_limit') or DEFAULT_PAGINATED_SERVER_LIMIT
-
         self.request_timeout = init_config.get('request_timeout') or DEFAULT_API_REQUEST_TIMEOUT
 
+        self.exclude_network_id_rules = set([re.compile(ex) for ex in init_config.get('exclude_network_ids', [])])
+        self.exclude_server_id_rules = set([re.compile(ex) for ex in init_config.get('exclude_server_ids', [])])
+        self.include_project_name_rules = set([re.compile(ex) for ex in init_config.get('whitelist_project_names', [])])
+        self.exclude_project_name_rules = set([re.compile(ex) for ex in init_config.get('blacklist_project_names', [])])
```

Comment:
This seems odd to rely on hash of compile object. Could we form set of regex strings first?

Reply:
What do you mean?

Reply:
I mean here we're relying on a Python implementation detail of unique object hashes instead of hashing strings themselves.